src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,177 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public static class Node<T> {
+        private T items;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T items, Node<T> next) {
+            this.prev = prev;
+            this.items = items;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (head == null && tail == null) {
+            head = tail = node;
+        } else {
+            node.prev = tail;
+            tail.next = node;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Invalid index");
+        }
+
+        Node<T> newNode = new Node<>(null, value, null);
+
+        if (head == null) {
+            head = tail = newNode;
+        } else if (index == 0) {
+            newNode.next = head;
+            head.prev = newNode;
+            head = newNode;
+        } else if (index == size) {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        } else {
+            Node<T> current = head;
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
```

Comment:
![image](https://github.com/mate-academy/jv-linked-list/assets/20326858/dcd5b60d-3130-4fbe-9ad4-838428282d55)

Reply:
not fixed